var/spack/repos/builtin/packages/intel-mpi-benchmarks/package.py

```diff
@@ -18,74 +18,108 @@ class IntelMpiBenchmarks(MakefilePackage):
+    variant("mt", default=True, description="Build MPI1 benchmark")
+
+    def url_for_version(self, version):
+        if version <= Version("2019.1"):
+            url = "https://github.com/intel/mpi-benchmarks/archive/refs/tags/v{0}.tar.gz"
+        else:
+            url = "https://github.com/intel/mpi-benchmarks/archive/refs/tags/IMB-v{0}.tar.gz"
+        return url.format(version)
 
-        if "benchmark=mpi1" in spec:
-            make("IMB-MPI1")
-        elif "benchmark=ext" in spec:
-            make("IMB-EXT")
-        elif "benchmark=io" in spec:
-            make("IMB-IO")
-        elif "benchmark=nbc" in spec:
-            make("IMB-NBC")
-        elif "benchmark=p2p" in spec:
-            make("IMB-P2P")
-        elif "benchmark=rma" in spec:
-            make("IMB-RMA")
-        elif "benchmark=mt" in spec:
-            make("IMB-MT")
+    @property
+    def build_directory(self):
+        if self.spec.satisfies("@2018"):
+            return "src"
         else:
-            make("all")
+            return "."
+
+    @property
+    def build_targets(self):
+        spec = self.spec
+        targets = []
+        if "+mpi1" in spec:
+            targets.append("MPI1")
+        elif "+ext" in spec:
+            targets.append("EXT")
+        elif "+io" in spec:
+            targets.append("IO")
+        elif "+nbc" in spec:
+            targets.append("NBC")
+        elif "+p2p" in spec:
+            targets.append("P2P")
+        elif "+rma" in spec:
+            targets.append("RMA")
+        elif "+mt" in spec:
+            targets.append("MT")
+
+        if self.spec.satisfies("@2019:"):
+            targets = ["TARGET=" + target for target in targets]
+
+        return targets
+
+    def edit(self, spec, prefix):
+        env["CC"] = spec["mpi"].mpicc
+        env["CXX"] = spec["mpi"].mpicxx
 
     def install(self, spec, prefix):
         mkdir(prefix.bin)
 
-        if "benchmark=mpi1" in spec:
-            install("IMB-MPI1", prefix.bin)
-        elif "benchmark=ext" in spec:
-            install("IMB-EXT", prefix.bin)
-        elif "benchmark=io" in spec:
-            install("IMB-IO", prefix.bin)
-        elif "benchmark=nbc" in spec:
-            install("IMB-NBC", prefix.bin)
-        elif "benchmark=p2p" in spec:
-            install("IMB-P2P", prefix.bin)
-        elif "benchmark=rma" in spec:
-            install("IMB-RMA", prefix.bin)
-        elif "benchmark=mt" in spec:
-            install("IMB-MT", prefix.bin)
+        if self.spec.satisfies("@2018"):
+            build_directory = "src"
         else:
-            install("IMB-EXT", prefix.bin)
-            install("IMB-IO", prefix.bin)
-            install("IMB-MPI1", prefix.bin)
-            install("IMB-MT", prefix.bin)
-            install("IMB-NBC", prefix.bin)
-            install("IMB-P2P", prefix.bin)
-            install("IMB-RMA", prefix.bin)
+            build_directory = "."
+
+        with working_dir(build_directory):
```

Comment:
Use self.build_directory and you can drop the duplicate if statement.

Reply:
Ok cool, thank you for the suggestion. I've made the change.